src/main/java/games/strategy/engine/framework/ui/background/WaitDialog.java

```diff
@@ -2,32 +2,39 @@
 
 import java.awt.BorderLayout;
 import java.awt.Component;
-import java.awt.event.ActionListener;
 
+import javax.annotation.Nullable;
 import javax.swing.JButton;
 import javax.swing.JDialog;
 import javax.swing.JOptionPane;
 
+import games.strategy.ui.SwingComponents;
+
 /**
  * A dialog that can be displayed during a long-running operation that optionally provides the user with the ability to
  * cancel the operation.
  */
 public final class WaitDialog extends JDialog {
   private static final long serialVersionUID = 7433959812027467868L;
 
-  public WaitDialog(final Component parent, final String waitMessage) {
-    this(parent, waitMessage, null);
+  public WaitDialog(final Component parent, final String message) {
+    this(parent, message, null);
   }
 
-  public WaitDialog(final Component parent, final String waitMessage, final ActionListener cancelAction) {
+  public WaitDialog(final Component parent, final String message, final @Nullable Runnable cancelAction) {
     super(JOptionPane.getFrameForComponent(parent), "Please Wait", true);
-    final WaitPanel panel = new WaitPanel(waitMessage);
-    getContentPane().setLayout(new BorderLayout());
-    getContentPane().add(panel, BorderLayout.CENTER);
+
+    setLayout(new BorderLayout());
+    add(new WaitPanel(message), BorderLayout.CENTER);
+
+    setDefaultCloseOperation(DO_NOTHING_ON_CLOSE);
     if (cancelAction != null) {
```

Comment:
And btw.: I think we should require action to be non null and pass an empty action instead.
If you insist on keeping null for this check, then nevermind.

Reply:
Unfortunately, passing an empty action would result in the Cancel button being created, which is probably not desirable.  However, I agree with the suggestion to not accept a `null` action for the cancelable case.  I think a little constructor refactoring can achieve this.

Reply:
Done in 90c642c.  I don't like the `Optional` parameter to the third constructor, but at least it's hidden as an implementation detail.  The only other solution I came up with required the duplication of a non-trivial line between the constructors, and I thought the private `Optional` usage was a better choice than the duplication.